src/models/DatasetModels.ts

```diff
@@ -16,6 +16,10 @@ export interface IValidator {
     validate(data: any, id?: string): ValidationStatus | Promise<ValidationStatus>;
 }
 
+export interface QValidator extends IValidator {
```

Comment:
We have already an `IValidator` right @pveleneni  - Can we re use that ?

Reply:
The IValidator allows only 1 method called validate, which is used in both Query Validator and Request validator. Need an additional method to validate query params, so added the interface. 

If not, have add params and check with an if condition for the type of validation to perform